virttest/utils_net.py

```diff
@@ -2688,7 +2688,14 @@ def __init__(self, params, vm_name):
                     existing_value = None
                 except IndexError:
                     existing_value = None
-                nic_dict[propertea] = nic_params.get(propertea, existing_value)
+                if propertea == "netdst" and existing_value is not None:
+                    if existing_value.split(':', 1)[0] == "shell":
+                        existing_value = decode_to_text(process.system_output(
+                            existing_value.split(':', 1)[1], shell=True))
+                    nic_dict[propertea] = existing_value
```

Comment:
Seems it change the original logic a little, when `propertea == "netdst" and existing_value is not None`, originally if `nic_params[propertea]` exist it will not use `existing_value`, but now it ignore `nic_params[propertea]`, not sure how other utils_net users use it, so I will prefer you keep original logic and only fit your own need when `existing_value.split(':', 1)[0] == "shell"`

Reply:
Also need check if `existing_value` you get is expected, the he code might be:
```
nic_dict[propertea] = nic_params.get(propertea, existing_value)
if propertea == "netdst" and existing_value is not None:
    if existing_value.split(':', 1)[0] == "shell":
       existing_value = process.getoutput(existing_value.split(':', 1)[1], shell=True)
       if existing_value is not None:
           nic_dict[propertea] = existing_value
       else:
           # raise error here since the shell command might not be correct
```

Reply:
Or try this:
```
if propertea == "netdst" and existing_value is not None:
    if existing_value.split(':', 1)[0] == "shell":
       existing_value = process.getoutput(existing_value.split(':', 1)[1], shell=True)
       if existing_value is not None:
           nic_dict[propertea] = existing_value
       else:
           # raise error here since the shell command might not be correct
       continue
nic_dict[propertea] = nic_params.get(propertea, existing_value)
```

Reply:
@vivianQizhu , could you hep review, I have updated code and test results in bz.